lib/msf/base/serializer/readable_text.rb

```diff
@@ -859,8 +860,10 @@ def self.dump_sessions(framework, opts={})
           'Header' => "Active sessions",
           'Columns' => columns,
           'Indent' => indent)
-      framework.sessions.each_sorted { |k|
-        session = framework.sessions[k]
+
+      framework.sessions.each { |k|
+        next unless session_ids.nil? || session_ids.include?(k[0])
+        session = k[1]
         row = create_msf_session_row(session, show_extended)
         tbl << row
       }
```

Comment:
@adfoster-r7 this conflicts with going back to each_sorted as that only iterates on the ids - thoughts on where to go?

Reply:
We didn't realise that `each_sorted` only gave session ids rather than the sessions themselves 
Taking a look at the `each` function though shows that it is sorted anyway
https://github.com/rapid7/metasploit-framework/blob/2f0509f2b56dbb73f28a19d5956de22c0bde0e91/lib/msf/core/session_manager.rb#L191-L197

so IMO I think we're good to stick with `each` since you need the session objects anyway and it keeps the same sorting